chia/full_node/full_node.py

```diff
@@ -1257,23 +1279,35 @@ async def add_block_batch(
         all_blocks: List[FullBlock],
         peer_info: PeerInfo,
         fork_info: Optional[ForkInfo],
+        current_ssi: uint64,
+        current_difficulty: uint64,
+        prev_ses_block: Optional[BlockRecord] = None,
```

Comment:
rather than passing these in as separate arguments and then return them as separate return values, you might want to consider adding these fields to `ForkInfo` instead. It's already used to pass in state of the current peak (be it a fork or the main chain) and this seems like a good fit. The fields are updated in-place in the `ForkInfo` object,

You won't need to pass in 3 additional parameters, nor return 3 additional ones

Reply:
i agree but there are all sort of cases (mainly because of the test code) and i rather merge this first before i try and consolidate into one object.

main issue is there are places where these are mandatory and places where they are not (iirc ForkInfo is always optional) and i want to get to a state where they are always mandatory and reduce the number of flows